### PR TITLE
Updating node group mappings to use an openshift specific tag.

### PR DESCRIPTION
**Pull request by abutcher**
Supersedes #8736

### CODE REVIEW

File: roles/openshift_aws/tasks/setup_master_group.yml [hunk 1]
```diff
@@ -26,7 +26,7 @@
     groups: "{{ openshift_aws_masters_groups }}"
     name: "{{ item.public_dns_name }}"
     hostname: "{{ openshift_aws_clusterid }}-master-{{ item.instance_id[:-5] }}"
-    openshift_node_group_name: "{{ openshift_aws_node_group_mappings[item.tags['sub-host-type']] }}"
+    openshift_node_group_name: "{{ item.tags['openshift-node-group-config'] }}"
```
Comment (kwoodson) [inline roles/openshift_aws/tasks/setup_master_group.yml]: These might need to be `|default('unset')` as well.

File: roles/openshift_aws/tasks/setup_scale_group_facts.yml [hunk 2]
```diff
@@ -46,5 +46,5 @@
     ansible_ssh_host: "{{ item.public_dns_name }}"
     name: "{{ item.public_dns_name }}"
     hostname: "{{ item.public_dns_name }}"
-    openshift_node_group_name: "{{ openshift_aws_node_group_mappings[item.tags['sub-host-type']] }}"
+    openshift_node_group_name: "{{ item.tags['openshift-node-group-config'] }}"
```
Comment (kwoodson) [inline roles/openshift_aws/tasks/setup_scale_group_facts.yml]: These might need to be `|default('unset')` as well.

File: roles/openshift_aws/tasks/setup_master_group.yml (hunk 1 above)
Comment (abutcher) [reply roles/openshift_aws/tasks/setup_master_group.yml]: :+1:

**Pull request merged**